lib/ansible/modules/system/hosts.py

```diff
@@ -0,0 +1,345 @@
+        for idx, line in enumerate(self.lines):
+            if (
+                    line['valid'] and
+                    line['ip'] == self.params['ip'] and (
+                        self.params['hostname'] is None or
+                        line['hostname'] == self.params['hostname'])):
+                if action == 'add':
+                    found = idx
+
+                    break
+                else:
+                    if found is None:
+                        found = []
+
+                    found.append(idx)
+
+        if action == 'add':
+            if found is None:
+                # If not found, add it
+                if not self.module.check_mode:
+                    record = self._format_line()
+
+                    self.lines.append({'line': record})
+
+                changed = True
+            elif self.params['alias'] is not None:
+                sorted_aliases_new = sorted(self.params['alias'])
+                sorted_aliases_old = sorted(self.lines[found]['alias'])
+
+                # Check if alias has changed
+                if sorted_aliases_new != sorted_aliases_old:
+                    self.lines[found]['alias'] = self.params['alias']
+                    self.lines[found]['line'] = self._format_line()
+
+                    changed = True
+
+        elif action == 'remove' and found is not None:
+            # If found, remove it
+            if not self.module.check_mode:
+                for idx in reversed(found):
+                    self.lines.pop(idx)
+
+            changed = True
+
+        if changed:
+            self._write_file()
+
+        return changed
+
+    def _write_file(self):
+        # Write into tmp file
+        tpm_fd, tmp_file = tempfile.mkstemp()
+
+        # Open the hosts file
+        try:
+            f = open(tmp_file, 'wb')
+        except IOError:
+            e = get_exception()
+            self.module.fail_json(
+                msg=(
+                    "Cannot open temporary hosts file %s for writing." %
+                    tmp_file),
+                details=str(e))
+
+        # Write data back into the file
+        for line in self.lines:
+            f.write(to_bytes(line['line']))
+            self.diff['after'] += line['line']
+
+        # Close the file
+        try:
+            f.close()
+        except IOError:
+            e = get_exception()
+            self.module.fail_json(
+                msg=(
+                    "Cannot close the temporal hosts file %s after writing." %
+                    tmp_file),
+                detail=str(e))
+
+        # Move the tmp file to the right location
+        self.module.atomic_move(
+            tmp_file, self.params['path'],
+            unsafe_writes=self.params['unsafe_writes'])
+
+
+def main():
+    # Module settings
+    module = AnsibleModule(
+        argument_spec=dict(
+            ip=dict(required=True),
+            hostname=dict(),
+            alias=dict(type='raw'),
```

Comment:
list might be a better type and then just leave off the space separated feature (comma separated should work IIRC)